transport/src/main/java/io/netty/channel/DefaultMaxMessagesRecvByteBufAllocator.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2015 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.channel;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufAllocator;
+
+public abstract class DefaultMaxMessagesRecvByteBufAllocator implements MaxMessagesRecvByteBufAllocator {
+    private volatile int maxMessagesPerRead;
+
+    public DefaultMaxMessagesRecvByteBufAllocator() {
+        // TODO(scott): this was previously based upon channel type, do we still need to consider channel type?
```

Comment:
@trustin - I don't think adding a protected abstract method to the config fits in with the intended use case.  The user should have the flexibility to swap in what ever `RecvByteBufAllocator` they want independent of the channel or configuration.  I will add a field to `ChannelMetadata` to enforce a minimum value instead, and let plz let me know what you think.